Modules/_json.c

```diff
@@ -1319,14 +1319,47 @@ encoder_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
         return NULL;
     }
 
+    /* Convert indent to str if it's not None or already a string */
+    if (indent != Py_None && !PyUnicode_Check(indent)) {
+        Py_ssize_t indent_level;
+        if (!PyIndex_Check(indent)) {
+            PyErr_Format(PyExc_TypeError,
+                         "make_encoder() argument 4 must be str, int, or None, "
+                         "not %.200s", Py_TYPE(indent)->tp_name);
+            return NULL;
+        }
+        indent_level = PyNumber_AsSsize_t(indent, PyExc_ValueError);
+        if (indent_level == -1 && PyErr_Occurred()) {
+            return NULL;
+        }
+        if (indent_level < 0) {
+            PyErr_SetString(PyExc_ValueError,
+                            "make_encoder() argument 4 must be a non-negative int");
+            return NULL;
+        }
+        /* Create a string of spaces: ' ' * indent_level */
+        indent = PyUnicode_New(indent_level, ' ');
+        if (indent == NULL) {
+            return NULL;
+        }
+        if (indent_level > 0) {
+            memset(PyUnicode_1BYTE_DATA(indent), ' ', indent_level);
+        }
```

Comment:
```suggestion
        memset(PyUnicode_1BYTE_DATA(indent), ' ', sizeof(Py_UCS1) * indent_level);
```

`memset` allows for a 0 size (and does nothing) as long as the pointer is valid. Alternatively, we could use `PyUnicode_Fill` but I think we usually bypass this and use `memset` directly.